movie-browser/src/App.js

```diff
@@ -1,31 +1,61 @@
 /* eslint-disable no-unused-vars */
 import React from 'react'
-import CardContainer from './CardContainer'
+import CardContainer from './view/CardContainer'
+import FeaturedMovie from './view/FeaturedMovie'
 import ApiReader from './model/ApiReader';
 
 // *** constant(s)
 const url = 'https://ghibliapi.herokuapp.com/films'
+const btnTxtDef = "View all"
+
+// TODO: fix sorting by year. have no idea why it works for score but not for year
+// TODO: makes button working
 
 // *** Actual App
 class App extends React.Component {
-  state = {movies: []};
+  state = {movies: [], featuredMovieIndex: 0, btnTxt: btnTxtDef};
 
   // *** running ApiReader when mounted
   componentDidMount() {
     this.fetchData();
   }
 
+  pickRandomMovieIndex = () => this.setState({
+    featuredMovieIndex: Math.floor(Math.random() * this.state.movies.length)
+  })
+
+  handleClickButton = () => {
+    if (this.state.btnTxt === btnTxtDef) {
+      this.setState({
+        btnTxt: "Hide all"
+      })
+    } else {
+      this.setState({
+        btnTxt: btnTxtDef
+      })
+    }
+    
+  }
+
   async fetchData() {
-    const data = await ApiReader(url);
+    const data = await ApiReader(url)
+
+    data.sort((a, b) => b.rt_score - a.rt_score)
```

Comment:
this should be done on the apiReader

Reply:
Extracted. Another 'thinking too much'... I thought I might need to use ApiReader.js for another purpose.